packages/app/src/cli/services/app-logs/logs-command/ui/components/hooks/usePollAppLogs.ts

```diff
@@ -52,7 +52,7 @@ export function usePollAppLogs({initialJwt, filters, resubscribeCallback}: UsePo
         }
         retryIntervalMs = result.retryIntervalMs
       } else {
-        setErrors([])
+        setErrors((errors) => (errors.length ? [] : errors))
```

Comment:
I don't really get how this fixed the issue?
Before, we were always setting the state to `[]` which caused the refresh.
Now, we're either setting the state to `[]` or `errors`, either of which should cause the refresh again?

I did some searching of the [useState docs](https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state), but it's not clear to me if providing a "updater" function also gets us a behaviour of "if the state hasn't changed, then don't update it".

Reply:
React re-renders if a state changes, and the way it determines if the state changed is by reference. When we `setState([])`, we're setting the state with a new array. Although the contents are the same, the reference is not, so React re-renders, causing the flickering.

To avoid this, we want to reset the errors only if there are already errors. Ideally, we'd do something like this:

```typescript
if (errors.length) {
  setErrors([])
}
```

But the problem with that is that `errors` inside this function actually refers to the initial `errors` array, not the current value! So we can't simply reset the errors based on the current `errors` value.

Luckily, React exposes the current value through the setter updater function. I use that current value to decide what we should do. If there are errors, then we set the state to `[]`, triggering a re-render. If there aren't errors, we set the state to the exact same reference, which does not trigger a re-render.